Update versions of dependencies for Bumblebee

osuleymanova opened the pull request:
Kotlin and Gradle versions updated
compileSdk and targetSdk versions updated
android:exported="true" in Manifest set
libraries version numbers updated
jcenter() replaced with mavenCentral()
typo "kotlin_versionsion" fixed

osuleymanova committed:
Update versions of dependencies for AFox

Kotlin and Gradle versions updated
compileSdk and targetSdk versions updated
android:exported="true" in Manifest set
libraries version numbers updated
jcenter() replaced with mavenCentral()
typo "kotlin_versionsion" fixed

osuleymanova committed:
Some updates

osuleymanova committed:
Some updates

wojtek-kalicinski commented on ktx-library-codelab/build.gradle:
```diff
@@ -45,23 +43,23 @@ buildscript {
+        mavenCentral()
         maven {
-            url "http://dl.bintray.com/kotlin/kotlin-eap"
+            url "https://dl.bintray.com/kotlin/kotlin-eap"
```
can you check if it will work without this? It might not be needed here and below

osuleymanova replied:
Hi Wojtek, the build failed with an error message regarding unsecured protocol.

wojtek-kalicinski replied:
What I meant was to remove that entry completely

osuleymanova replied:
Removed.

wojtek-kalicinski commented on ktx-library-codelab/build.gradle:
```diff
@@ -45,23 +43,23 @@ buildscript {
+        mavenCentral()
         maven {
-            url "http://dl.bintray.com/kotlin/kotlin-eap"
+            url "https://dl.bintray.com/kotlin/kotlin-eap"
```
might not be needed anymore?

osuleymanova replied:
The same as for above: on my side it works with secured protocol only.

osuleymanova replied:
Removed.

wojtek-kalicinski commented on ktx-library-codelab/step-06/myktxlibrary/src/main/java/com/example/android/myktxlibrary/LocationUtils.kt:
```diff
@@ -58,7 +58,7 @@ fun FusedLocationProviderClient.locationFlow() = callbackFlow<Location> {
             for (location in result.locations) {
                 try {
-                    offer(location) // emit location into the Flow using ProducerScope.offer
+                    trySend(location).isSuccess // emit location into the Flow using ProducerScope.offer
```
`isSuccess` doesn't do anything here (the result is dropped anyway), so maybe we could remove it?

osuleymanova replied:
There was an error message: Using 'offer(E): Boolean' is an error. Deprecated in the favour of 'trySend' method. It has proven itself as the most error-prone method in Channel API.

wojtek-kalicinski replied:
I agree with the change to trySend(location), I was only asking to remove isSuccess

osuleymanova replied:
Sorry for misunderstanding.

osuleymanova replied:
iSuccess removed.

wojtek-kalicinski commented on ktx-library-codelab/step-06/myktxlibrary/src/main/java/com/example/android/myktxlibrary/LocationUtils.kt:
```diff
@@ -58,7 +58,7 @@ fun FusedLocationProviderClient.locationFlow() = callbackFlow<Location> {
             result ?: return
             for (location in result.locations) {
                 try {
```
I think with `trySend` instead of `offer` the try/catch is not needed, as trySend will not throw an exception

wojtek-kalicinski replied:
I meant removing the try/catch and just leaving what's inside, like this:
```
override fun onLocationResult(result: LocationResult?) {
  result ?: return
  for (location in result.locations) {
     trySend(location)
  }
}
```

osuleymanova replied:
Done.

osuleymanova committed:
isSuccess removed, some other updates

osuleymanova committed:
Update LocationUtils.kt

osuleymanova committed:
Build.gradle updates

AGP and gradle versions updated

osuleymanova committed:
Gradle and dependencies versions update

Kotlin version updated
SDK versions updated
Gradle and AGP versions updated
Dependencies versions updated
